src/Illuminate/Http/Request.php

```diff
@@ -352,7 +352,9 @@ public function only($keys)
         $input = $this->all();
 
         foreach ($keys as $key) {
-            Arr::set($results, $key, data_get($input, $key));
+            if (array_key_exists($key, $input)) {
```

Comment:
Test fail, the `$key` support dot notation. You should use `Arr::has` instead of `array_key_exists`

Reference: #12651 `Arr::only`